server/lib/queries.js

```diff
@@ -1017,9 +1021,13 @@ const getTaxFormsRequiredForExpenses = async expenseIds => {
       ON all_expenses_collectives.id = all_expenses."CollectiveId"
       AND all_expenses_collectives."HostCollectiveId" = d."HostCollectiveId"
     LEFT JOIN "LegalDocuments" ld
-      ON ld."CollectiveId" = analyzed_expenses."FromCollectiveId"
-      AND ld.year + :validityInYears >= date_part('year', analyzed_expenses."incurredAt")
+      ON ld.year + :validityInYears >= date_part('year', analyzed_expenses."incurredAt")
       AND ld."documentType" = 'US_TAX_FORM'
+      AND ld."requestStatus" = 'RECEIVED'
```

Comment:
Here I'm filtering to get only the received documents to ignore all expenses when a doc is available - either on the account or the host - with the `AND ld.id IS NULL` below.